### DISCUSSION TITLE
Help with Continuous Deployment (PTE with AppSource Library Dependency)

### DISCUSSION THREAD

**Discussion by jngillard** [microsoft/AL-Go#1028] [Q&A] [answered]
Is it possible to setup continuous Deployment to a SAAS sandbox environment where the App has a dependency to another app (in a separate repository) where that app is an AppSource library app? The CI/CD fails at the deployment step where it tries to install the App. It fails because the dependent library app is not already installed in the environment.

I don't know why it doesn't try and install the dependent app first. Anny assistance would be appreciated.
Thanks,

**Comment by freddydk** [comment]
Before talking about continuous deployment, what happens when you try to install the PTE manually in a new tenant where the AppSource library app isn't installed?
Does that work? will it pull in the AppSource library app automagically?

**Reply by jngillard** [reply]
If I try and publish this app directly from VSCode to the tenant sandbox environment, I run into the same error I get when the build task runs. If I compile this app as a PTE and install from extension management within Business Central, it pulls the appsource library dependent application automatically and installs the PTE correctly. Now, since the library app is indeed published in the tenant environment, the deploy step in AL-GO succeeds.

**Comment by freddydk** [comment, marked as answer]
The latest version of AL-Go for GitHub has a property called DependencyInstallMode, which defaults to "install". It seems like this is exactly what you need here.
https://aka.ms/algosettings#dependencyinstallmode
